openedx/core/djangoapps/agreements/api.py

```diff
@@ -68,3 +73,180 @@ def get_integrity_signatures_for_course(course_id):
+    except ObjectDoesNotExist:
+        #create
+        signature = LTIPIISignature.objects.create(
+            user=user,
+            course_key=course_key,
+            lti_tools=lti_tools,
+            lti_tools_hash=lti_tools_hash)
+        return signature
+    else:
+        signature = LTIPIISignature.objects.update(
+            user=user,
+            course_key=course_key,
+            lti_tools=lti_tools,
+            lti_tools_hash=lti_tools_hash)
+        return signature
+
+
+def get_lti_pii_signature(username, course_id):
+    """
+    Get the lti pii signature of a user in a course.
+
+    Arguments:
+        * username (str)
+        * course_id (str)
+
+    Returns:
+        * An LTIPIISignature object, or None if one does not exist for the
+          user + course combination.
+    """
+    course_key = CourseKey.from_string(course_id)
+    try:
+        user = User.objects.get(username=username)
+        signature = LTIPIISignature.objects.get(user=user, course_key=course_key)
+    except ObjectDoesNotExist:
+        return None
+    else:
+        return LTIPIISignatureData(user=signature.user, course_id=str(signature.course_key),
+                                   lti_tools=signature.lti_tools, lti_tools_hash=signature.lti_tools_hash)
+
+
+def get_pii_receiving_lti_tools(course_id):
+    """
+    Get a course's LTI tools that share PII.
+
+    Arguments:
+        * course_id (str)
+
+    Returns:
+        * A List of LTI tools sharing PII.
+    """
+
+    course_key = CourseKey.from_string(course_id)
+    try:
+        course_ltipiitools = LTIPIITool.objects.get(course_key=course_key).lti_tools
+        return LTIToolsReceivingPIIData(
+            lii_tools_receiving_pii=course_ltipiitools,
+        )
+    except ObjectDoesNotExist:
+        return None
+
+
+def user_lti_pii_signature_needed(username, course_id):
+    """
+    Determines if a user needs to acknowledge the LTI PII Agreement
+
+    Arguments:
+        * username (str)
+
+    Returns:
+        * True if the user needs to sign a new acknowledgement.
+        * False if the acknowledgements are up to date.
+    """
+    if _course_has_lti_pii_tools(course_id):
+        if _user_lti_pii_signature_exists(username, course_id):
+            if _user_needs_signature_update(username, course_id):
+                # up to date
+                return False
+            else:
+                # lti pii signature needs to be updated
+                return True
+        else:
+            # write a new lti pii signature
+            return True
+    else:
+        return False
+
+
+def _course_has_lti_pii_tools(course_id):
+    """
+    Determines if a specifc course has lti tools sharing pii
+
+    Arguments:
+        * course_id (str)
+
+    Returns:
+        * True if the course does have a list.
+        * False if the course does not.
+    """
+    course_key = CourseKey.from_string(course_id)
+    try:
+        LTIPIITool.objects.get(course_key)
```

Comment:
```suggestion
        LTIPIITool.objects.get(course_key=course_key)
```

Does this work? I think you need to specify the model field against which you're doing the match on `course_key`. Or does this work because `course_key` is the primary key?

Reply:
Nope, this was a mistype on my end.